.github/workflows/reusable_deploy_v2_sar.yml

```
@@ -0,0 +1,124 @@
+name: Deploy V2 SAR
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  SAM_VERSION: 2.44.0
+  NODE_VERSION: 16.12
+  AWS_REGION: eu-west-1
+  SAR_NAME: aws-lambda-powertools-python-layer-v2
+
+on:
+  workflow_call:
+    inputs:
+      stage:
+        description: "Deployment stage (BETA, PROD)"
+        required: true
+        type: string
+      artefact-name:
+        description: "CDK Layer Artefact name to download"
+        required: true
+        type: string
+      package-version:
+        description: "The version of the package to deploy"
+        required: true
+        type: string
+      environment:
+        description: "GitHub Environment to use for encrypted secrets"
+        required: true
+        type: string
+
+jobs:
+  deploy-sar-app:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+    strategy:
+      matrix:
+        architecture: ["x86_64", "arm64"]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ secrets.AWS_LAYERS_ROLE_ARN }}
+      - name: AWS credentials SAR role
+        uses: aws-actions/configure-aws-credentials@v1
+        id: aws-credentials-sar-role
+        with:
+          aws-access-key-id: ${{ env.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          aws-session-token: ${{ env.AWS_SESSION_TOKEN }}
+          role-duration-seconds: 1200
+          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ secrets.AWS_SAR_V2_ROLE_ARN }}
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.artefact-name }}
+      - name: Unzip artefact
+        run: unzip cdk.out.zip
+      - name: Configure SAR name
+        run: |
+          NAME=${{ env.SAR_NAME }}
+          if [[ "${{ inputs.stage }}" == "BETA" ]]; then
+            NAME="test-$NAME"
```

```suggestion
            NAME="test-${NAME}"
```

guard from expansion